examples/correlation_matrices/sampler.cpp

```diff
@@ -147,6 +149,46 @@ void correlation_matrix_uniform_sampling_MT(const unsigned int n, const unsigned
     write_to_file<PointMT>(walkname + "_matrices_MT" + std::to_string(n) + ".txt", randPoints);
 }
 
+template<typename WalkTypePolicy, typename PointType, typename RNGType, typename MT>
+void tune_walkL(const std::vector<unsigned int>& walkL_values, const std::vector<unsigned int>& dimensions, const unsigned int num_points,
+            	const unsigned int nburns, const unsigned int num_matrices){
+    for (unsigned int n : dimensions) {
+    	std::list<MT> randCorMatrices;
+   	 
+    	for (unsigned int walkL : walkL_values) {
+            std::chrono::steady_clock::time_point start, end;
+            double time;
+            start = std::chrono::steady_clock::now();
+
+            uniform_correlation_sampling_MT<WalkTypePolicy, PointType, RNGType>(n, randCorMatrices, walkL, num_points, 0);
+
+            end = std::chrono::steady_clock::now();
+            time = std::chrono::duration_cast<std::chrono::milliseconds>(end - start).count();
+            std::cout << "Elapsed time : " << time << " (ms)" << std::endl;
+
+            VT samples(n, randCorMatrices.size());
+            int idx = 0;
+            for (const auto& matrix : randCorMatrices) {
```

Comment:
randCorMatrices is a list of matrices. However, psrf and ess need vectors to work.
We have to vectorize the matrix here: Store as a vector the upper (or lower) triangular part.
I think you had implemented a function for this in your previous PR, is that correct?